lib/bemhtml/compiler.js

```diff
@@ -134,7 +146,15 @@ Compiler.prototype.markMatches = function markMatches(ast) {
     };
   }
 
-  if (ast.type !== 'CallExpression') return ast;
+  if (ast.type !== 'CallExpression') {
+    if (ast.type === 'Identifier' &&
+        (ast.name === 'replace' || ast.name === 'extend')) {
+      // Desugar`replace' mode predicate and set the flag 
+      ast.extendedModeDetected = ast.name;
```

Comment:
Name `extendedModeDetected` suggests that the value is boolean, but in fact it is a mode itself. Why not just call it `extendedMode` then?